src/relax/analysis/well_formed.cc

```diff
@@ -260,6 +261,19 @@ class WellFormedChecker : public relax::ExprVisitor,
       }
     }
 
+    // call_tir works for tir PrimFunc, call_dps_packed works for ExternFunc
+    static const Op& call_tir_op = Op::Get("relax.call_tir");
+    static const Op& call_dps_packed_op = Op::Get("relax.call_dps_packed");
+    if (op->op == call_tir_op && !op->args[0]->IsInstance<GlobalVarNode>()) {
+      Malformed(Diagnostic::Error(op->args[0]->span)
+                << "call_tir expects a prim_func, but gets " << op->args[0]->GetTypeKey());
+    }
```

Comment:
As mentioned in the community meeting, we can enforce this in the `FInferStructInfo` for `call_tir` and not need any special cases in the well-formedness check.